sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -112,11 +119,19 @@
   private static final long MILLIS_PER_DAY = 86400000L;
 
   private static final ParameterExpression rowParam = Expressions.parameter(Row.class, "row");
+  private PTransform<PCollection<BeamCalcRelError>, POutput> errorsTransformer;
+  private static final TupleTag<Row> rows = new TupleTag<Row>() {};
+  private static final TupleTag<BeamCalcRelError> errors = new TupleTag<BeamCalcRelError>() {};
 
   public BeamCalcRel(RelOptCluster cluster, RelTraitSet traits, RelNode input, RexProgram program) {
     super(cluster, traits, input, program);
   }
 
+  @Override
+  public void withErrorsTransformer(PTransform<PCollection<BeamCalcRelError>, POutput> ptransform) {
```

Comment:
Should this be an argument to `buildPTransform` instead? It seems kind of roundabout to be doing it this way.

Reply:
Yes, I can add to `buildPTransform`, but it will break the API for all `BeamRelNode` implementations. 
When I add it with a default function, any implementation  that don't need to handle errors can continue as it was before, and the one that want to do error handling will implement this method and use this variable.

Reply:
You could also add a default for the new `buildPTransform` that calls the old one, this would allow you to avoid storing state and also not make a breaking change. There are other places the pipeline can crash, we'll eventually want every RelNode to catch and return errors. I'm fine with leaving it as is, this is an internal API so easy for someone to change again if it becomes a problem.

Reply:
good idea, this is easy fix 6bf54e2